src/components/Preferences/Availability/NumTeachingDays.tsx

```diff
@@ -0,0 +1,52 @@
+import { Text, FormControl, FormLabel, HStack } from "@chakra-ui/react";
+
+import NumInput from "@components/NumInput";
+
+const NumTeachingDays = ({ values, setFieldValue, isDisabled = false }) => {
+    return (
+        <FormControl>
+            <FormLabel>Preferred Number of Teaching Days</FormLabel>
+            <HStack align="left">
+                <Text alignSelf="center">Fall</Text>
+                <NumInput
+                    name="teachingDaysPerWeek.fall"
+                    isDisabled={isDisabled || values.fall == 0}
```

Comment:
the or needs to check the 'Preferred Number of Teaching Courses per Semester' instead. Right now its checking its own value and gets disabled if you change it to 0